fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergPartitionUtils.java

```diff
@@ -131,4 +144,87 @@ public static Set<IcebergPartition> getAllPartition(Table table) {
         }
         return builder.build();
     }
+
+    // Normalize partition name to yyyy-MM-dd (Type is Date) or yyyy-MM-dd HH:mm:ss (Type is Datetime)
+    // Iceberg partition field transform support year, month, day, hour now,
+    // eg.
+    // year(ts)  partitionName : 2023              return 2023-01-01 (Date) or 2023-01-01 00:00:00 (Datetime)
+    // month(ts) partitionName : 2023-01           return 2023-01-01 (Date) or 2023-01-01 00:00:00 (Datetime)
+    // day(ts)   partitionName : 2023-01-01        return 2023-01-01 (Date) or 2023-01-01 00:00:00 (Datetime)
+    // hour(ts)  partitionName : 2023-01-01-12     return 2023-01-01 12:00:00 (Datetime)
+    public static String normalizeTimePartitionName(String partitionName, PartitionField partitionField, Schema schema,
+                                                    Type type) {
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        boolean parseFromDate = true;
+        if (partitionField.transform().toString().equalsIgnoreCase("year")) {
+            partitionName += "-01-01";
+        } else if (partitionField.transform().toString().equalsIgnoreCase("month")) {
+            partitionName += "-01";
+        } else if (partitionField.transform().toString().equalsIgnoreCase("day")) {
+            dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        } else if (partitionField.transform().toString().equalsIgnoreCase("hour")) {
+            dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd-HH");
+            parseFromDate = false;
+        }
+
+        // partition name formatter
+        DateTimeFormatter formatter = null;
+        if (type.isDate()) {
+            formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        } else {
+            formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+        }
+        // If has timestamp with time zone, should compute the time zone offset to UTC
```

Comment:
why use the UTC timezone instead of system timezone ?

Reply:
Because for iceberg, the type timestamp with time zone is store as UTC 
![image](https://github.com/StarRocks/starrocks/assets/9495145/07637c16-275f-4ba4-8ac2-6a7b9627f56b)